util/gdb_funcs.py

```diff
@@ -0,0 +1,167 @@
+    translation=0x55555facf4b0, mode=gem5::BaseMMU::Execute) at src/arch/arm/mmu.hh:251
+251         translateTiming(const RequestPtr &req, ThreadContext *tc,
+(gdb) print _name
+$1 = "board.processor.cores0.core.mmu"
+"""
+
+import re
+
+
+# NOTE: no need to import anything since gdb has its own Python environment.
+# Additionally, this script only works within gdb.
+class TargetNamedBreakpoint(gdb.Breakpoint):
+    def __init__(
+        self, location: str, expose_gdb_exceptions: bool = False
+    ) -> None:
+        """
+        This class allows the user to create Breakpoint objects that only stop
+        when their code location is reached for certain Named objects.
+        The list of qualified Named objects could be expressed either as exact
+        matches or regex matches. Exact matches could be added to the breakpoint
+        through `add_match_exact` method and regex matches could be added
+        `add_match_regex` method.
+
+        :param location: The location where the breakpoint should be set. E.g.
+            "gem5::ruby::Cache_Controller::AllocateTBE_SeqRequest"
+        :param expose_gdb_exceptions: To determine whether gdb needs to stop upon
+        hitting an object of this class, this class overrides the `stop` method.
+        This breakpoint will search for _name within the context of the breakpoint.
+        It should succeed assuming the breakpoint is set within the scope of a
+        Named class, which SimObject and its children are.
+        If the breakpoint fails at extracting _name from the context,
+        it will raise exceptions.
+        This class treats gdb related exceptions separately from other exceptions.
+        The developer suspects all the gdb exceptions to be related to the
+        extraction of _name from the context.
+        If this parameter is set to True, the class will raise gdb exceptions so
+        that the user can course correct.
+        If this parameter is set to False, the class will print the error message
+        and continue without stopping.
+        """
+        super().__init__(location)
+        self._expose_gdb_exceptions = expose_gdb_exceptions
+        self._exact_matches = set()
+        self._regex_matches = set()
+
+    def add_match_exact(self, exact_match: str) -> None:
+        """
+        Add an exact match to the list of potential names.
+
+        :param exact_match: The name to exact_match against upon hitting the breakpoint.
+        E.g. "system.cache_hierarchy.core_clusters0.dcache" if you want to stop
+        only for dcache for core_clusters0.
+        """
+        self._exact_matches.add(exact_match)
+
+    def add_match_regex(self, regex_match: str) -> None:
+        """
+        Add a regex match to the list of potential names.
+
+        :param regex_match: The pattern to regex_match against upon hitting the breakpoint.
+        E.g. "system.cache_hierarchy.core_clusters(0-9+).dcache" if you want to stop
```

Comment:
```suggestion
        E.g. "system.cache_hierarchy.core_clusters[0-9]+.dcache" if you want to stop
```
Hey Mahyar, sorry about the nitpick. Should this be "system.cache_hierarchy.core_clusters[0-9]+.dcache" instead? I tried testing this change with `python test.add_match_regex("board.processor.cores(0-9+).core.mmu")` (i.e. modifying the example at the top of the file) and didn't hit a breakpoint.

Reply:
@mahyarsamani can you take a look at this feedback?